boofuzz/sessions.py

```diff
@@ -1021,54 +1107,39 @@ def _restart_target(self, target):
         @raise exception.BoofuzzRestartFailedError if restart fails.
         """
 
+        # TODO: reuse_target_connection seems to be only handled when using
+        #       a custom callback. wtf?
+
         self._fuzz_data_logger.open_test_step("Restarting target")
         if len(self.on_failure) > 0:
             for f in self.on_failure:
                 self._fuzz_data_logger.open_test_step("Calling registered on_failure method")
                 f(logger=self._fuzz_data_logger)
```

Comment:
Honestly, it did not even occur to me that this is a callback that could end up there. as far as I see it, it isn't even possible to set it from the constructor, it can only be added using the EventHook stuff when the Session was instantiated.

To be completely honest, I'm inclined to remove it entirely, as it's basically the same as restart_target, and very misleading as it isn't only called when a failure occurs but also when for example a restart threshold is reached.

Another possibility would be to move the code to _process_failures and leave it as-is, keeping the hook-on-failure aspect. That could then theoretically be put as a monitor duty as well.

Reply:
Ah OK, let's leave it as is for expediency. After an issue search, I see a related problem (?) was recently discussed: https://github.com/jtpereyda/boofuzz/issues/390

Should probably be sorted out, but let's just call it a separate concern for now.